package/batocera/core/batocera-scripts/scripts/batocera-create-collection

```diff
@@ -0,0 +1,170 @@
+  add gb games with "batocera-create-collection -s gb zelda".
+
+Gotchas: 
+  - 'search_terms' for games are case insensitive...
+  - ... but 'genres' and 'systems' are case sensitive (blame scapers disparities)
+
+EOF
+}
+
+# Does a list contain an element?
+contains_element() {
+	local e match="$1"
+	shift
+	for e; do 
+		[[ "$e" == "$match" ]] && return 0
+	done
+	return 1
+}
+
+# Look for ROM suitable extensions
+build_ext_array() {
+	[ -f ${user_systemscfg} ] && systemscfg=${user_systemscfg}
+	all_ext=$(xmllint --xpath '/systemList/system/extension/text()' ${systemscfg} 2>/dev/null)
+	ext_array="" 
+	for i in ${all_ext}; do
+		! contains_element ${i} ${ext_array} && ext_array="$i $ext_array" 
+	done
+	extensions="("$(echo ${ext_array}|sed "s/ /|/g")")"
+}
+
+# First, let's do a match with ROM filenames
+# We'll parse only roms from matching extensions
+parse_filenames() {
+	fn=$(date +"%s")
+	tmpfile=/tmp/collec_${fn}
+	depth=1
+	[[ -z $systempath ]] && depth=2
+	find ${rompath}/${systempath} -type f -maxdepth ${depth} | egrep -i "${rompath}.*${pattern}.*${extensions}" > ${tmpfile}
+	while IFS= read -r game; do 
+		! fgrep -e "${game}" ${filecollection} >/dev/null 2>/dev/null && echo ${game} >> ${filecollection} && new_found=$((new_found+1))
+	done < ${tmpfile}
+	[[ -f $tmpfile ]] && rm ${tmpfile}
+}
+
+# Now, let's explore the scraped metadata (in particular for arcade, neogeo...)
+# Uppercase only the first letter, as it's the default format for scrapers
+parse_gamelists() {
+	lowerpattern=$(echo $pattern | awk '{print tolower($0)}' | awk -v FS='.' -v OFS=' ' '{for(i=1;i<=NF;i++){$i=toupper(substr($i,1,1)) substr($i,2)}}1')
+	fn=$(date +"%s")
+	tmpfile=/tmp/collec_${fn}
+	depth=0
+	[[ -z $systempath ]] && depth=1
+	systemslist=$(find ${rompath}/${systempath} -type d -maxdepth ${depth})
+	for system in ${systemslist}; do
+		xmltmp=${system}/gamelist.xml 
+		[ -f ${xmltmp} ] || continue
+		if ! [[ -z $lowerpattern ]]; then
+			if ! [[ -z $genre ]]; then
+				xmllint --xpath "/gameList/game[contains(name, \"${lowerpattern}\") and contains(genre, \"${genre}\")]/path/text()" ${xmltmp} 2>/dev/null | sed 's:^\./::' > ${tmpfile}
+			else
+				xmllint --xpath "/gameList/game[contains(name, \"${lowerpattern}\")]/path/text()" ${xmltmp} 2>/dev/null | sed 's:^\./::' > ${tmpfile}
+			fi
+		else
+			xmllint --xpath "/gameList/game[contains(genre, \"${genre}\")]/path/text()" ${xmltmp} 2>/dev/null | sed 's:^\./::' > ${tmpfile}
+
+		fi
+		while IFS= read -r game; do 
+			if ! fgrep -e "${system}/${game}" ${filecollection} >/dev/null 2>/dev/null; then
+				[[ -f $system/$game ]] && echo ${system}/${game} >> ${filecollection} && new_found=$((new_found+1))
+			fi
+		done < ${tmpfile}
+	done
+	[[ -f $tmpfile ]] && rm ${tmpfile}
+}
+
+# Main loop
+while getopts ":g:c:s:h" opt; do
+	case ${opt} in
+		h )
+			print_help
+			exit
+			;;
+		g )
+			genre=$OPTARG
+			;;
+		c )
+			collec=$OPTARG
+			;;
+		s )
+			systempath=$OPTARG
```

Comment:
```
systempath=${OPTARG,,}
```
will lead to a hit with ignoring mixed case

Reply:
ok, why not.